VideoLocker/src/main/java/org/edx/mobile/event/NewVersionAvailableEvent.java

```diff
@@ -0,0 +1,189 @@
+package org.edx.mobile.event;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.annotation.StringRes;
+import android.support.v4.util.ArrayMap;
+
+import org.apache.commons.lang3.BooleanUtils;
+import org.apache.commons.lang3.ObjectUtils;
+import org.apache.commons.lang3.time.FastDateFormat;
+import org.edx.mobile.R;
+import org.edx.mobile.third_party.versioning.ArtifactVersion;
+import org.edx.mobile.util.ResourceUtil;
+
+import java.text.DateFormat;
+import java.text.Format;
+import java.util.Date;
+import java.util.Map;
+
+import de.greenrobot.event.EventBus;
+
+/**
+ * An event signifying that a new version of the app is available on the app stores.
+ */
+public class NewVersionAvailableEvent implements Comparable<NewVersionAvailableEvent> {
+    /**
+     * Post an instance of NewVersionAvailableEvent on the event bus, based on the provided
+     * properties, if this hasn't been posted before. The sticky events will be queried for an
+     * existing report, and a new one will only be posted if it has more urgent information than the
+     * previous one. The events are posted and retained as sticky events in order to have a
+     * conveniently and semantically accessible session-based singleton of it to compare against,
+     * but this has the implication that they can't be removed from the event bus after consumption
+     * by the subscribers. To address this restriction, this class defined methods to mark instances
+     * as having being consumed, which can be used by subscribers for this purpose.
+     *
+     * @param newVersion        The version number of the latest release of the app.
+     * @param lastSupportedDate The last date on which the current version of the app will be
+     *                          supported.
+     * @param isUnsupported     Whether the current version is unsupported. This is based on whether
+     *                          we're getting HTTP 426 errors, and thus can't be inferred from the
+     *                          last supported date (the two properties may not be consistent with
+     *                          each other due to wrong local clock time or an inconsistency in the
+     *                          server configurations).
+     */
+    public static void post(@Nullable final ArtifactVersion newVersion,
```

Comment:
There are three properties here, each corresponding to one of the notification types that I described in the summary, and each also corresponding to a header, or an error code in the case of `isUnsupported`. Any of these may be `null` (or `false`), in which case it's automatically marked as a type corresponding to the property with the highest priority that is available. I suppose it would not be a valid event if all three properties are `null` or `false`, and I'll add a check for that in the constructor and here.